src/services/patchwork.rs

```diff
@@ -83,10 +83,20 @@ pub fn start<
                                 player_state.clone(),
                             )
                             .unwrap(),
-                            None => Anchor {
-                                conn_id: None,
-                                map_index: new_map_index,
-                            },
+                            None => {
+                                gameplay_router::route_packet(
+                                    msg.packet.clone(),
+                                    msg.conn_id,
+                                    player_state.clone(),
+                                );
```

Comment:
We need to process this packet locally before we can reintroduce the player to peers- that's because before this packet is processed, the player's position is still actually on the Peers map. This is the exact packet that takes them over the line. So if we want to send a spawn player packet with an accurate position, this packet needs to be processed first.